### PR TITLE
Add pyctest

### PR DESCRIPTION
**Pull request by jrmadsen**

### CODE REVIEW

**Comment by jrmadsen** [comment]
@conda-forge/staged-recipes I believe this recipe is ready. Thanks.

**Comment by jrmadsen** [comment]
@conda-forge-admin @conda-forge-manager Please merge PR

**Comment by CJ-Wright** [comment]
@conda-forge/staged-recipes

File: recipes/pyctest/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - pyctest.cmake
+    - pyctest.ctest
+    - pyctest.cpack
+
+about:
+  home: http://github.com/jrmadsen/pyctest
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: 'Python bindings to select portions of CMake/CTest package -- enabling generation of CTests for CDash submission without a CMake build system'
```
Comment (CJ-Wright) [inline recipes/pyctest/meta.yaml]: would it be possible to wrap this?

File: recipes/pyctest/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    - git
+    - python={{ python }}  # [win]
+  host:
+    - python={{ python }}
+    - pip
+    - setuptools
+    - openssl
+    - curl
+  run:
+    - python={{ python }}
```
Comment (CJ-Wright) [inline recipes/pyctest/meta.yaml]: Is this `python={{ python }}` syntax needed?

**Comment by jrmadsen** [comment]
Thanks

File: recipes/pyctest/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+  skip: true  # [vc<14] [win]
```
Comment (CJ-Wright) [inline recipes/pyctest/meta.yaml]: I think this needs to be `# [vc<14 and win]` (although we only use `vc` on windows so `win` might not be needed?)

Comment (jrmadsen) [reply recipes/pyctest/meta.yaml]: I added `# [win]` when it failed locally on Linux but I can try the and

File: recipes/pyctest/meta.yaml (hunk 2 above)
Comment (jrmadsen) [reply recipes/pyctest/meta.yaml]: Nope probably not

File: recipes/pyctest/meta.yaml (hunk 1 above)
Comment (jrmadsen) [reply recipes/pyctest/meta.yaml]: Sure

**Comment by jrmadsen** [comment]
@CJ-Wright I believe everything has been resolved now

File: recipes/pyctest/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - git
```
Comment (scopatz) [inline recipes/pyctest/meta.yaml]: Why do you need git here?

Comment (jrmadsen) [reply recipes/pyctest/meta.yaml]: The CMake build system checks if the submodule folders have anything in them and if they haven't been checked out, it does a `git submodule update --recursive <path>`.

**Pull request merged**